### PR TITLE
refactor: migrate debug api use case

### PR DESCRIPTION
**Pull request by ytvnr**
## Issue

https://gravitee.atlassian.net/browse/APIM-3441

## What's remaining

- [x] Create `Instance` in core model, map it in crud service
- [x] Merge `ApiPolicyValidator` class and interface
- [x] Remove `DebugEventDomainServiceLegacyWrapper` to only have the `EventCrudService` (+ create a `Event` model in core)
- [x] Instead of `DebugApiEntity`, use `DebugApi` (definition) as Input for the use case
   - [x] mapping should be done in the resource
   - [x] send execution mode from front
   - [x] we can probably avoid to fetch the whole api and use environment from audit info, apiId from input and so on.
- [x] ~pass null as ExecutionContext for EventService as it is not used by its implementation (add a comment explaining that). It avoid the use case to be polluted by the creation of an ExecutionContext~ Not possible because the `executionContext` is manipulated by the legacy implementation, so we need to keep an object; solution is to create it in the `LegacyWrapper` implementation.
- [x] Add a test case for `/_debug` in `ApiResourceTest`
- [x] Fix `CoreRulesTest`

🏗️ Your changes can be tested here and will be available soon:
      Console: [https://pr.team-apim.gravitee.dev/6049/console](https://pr.team-apim.gravitee.dev/6049/console)
      Portal: [https://pr.team-apim.gravitee.dev/6049/portal](https://pr.team-apim.gravitee.dev/6049/portal)
      Management-api: [https://pr.team-apim.gravitee.dev/6049/api/management](https://pr.team-apim.gravitee.dev/6049/api/management)
      Gateway v4: [https://pr.team-apim.gravitee.dev/6049](https://pr.team-apim.gravitee.dev/6049)
      Gateway v3: [https://pr.gateway-v3.team-apim.gravitee.dev/6049](https://pr.gateway-v3.team-apim.gravitee.dev/6049)

---

📚&nbsp;&nbsp;View the storybook of this branch [here](https://612657caa8e859003a8a6430-wmmggkjtzt.chromatic.com)

### CODE REVIEW

File: gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/infra/domain_service/api/ApiPolicyValidatorDomainServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+import java.util.Collection;
+import java.util.Set;
+import java.util.stream.Stream;
+import lombok.RequiredArgsConstructor;
+import org.jetbrains.annotations.Nullable;
+import org.springframework.stereotype.Service;
+
+@RequiredArgsConstructor
+@Service
+public class ApiPolicyValidatorDomainServiceImpl implements ApiPolicyValidatorDomainService {
```
Comment (ytvnr) [inline gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/infra/domain_service/api/ApiPolicyValidatorDomainServiceImpl.java]: merge with interface

File: gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/core/debug/domain_service/DebugEventDomainService.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+import io.gravitee.definition.model.debug.DebugApi;
+import io.gravitee.rest.api.model.EventEntity;
+import io.gravitee.rest.api.service.common.ExecutionContext;
+import java.util.Map;
+import java.util.Set;
+
+public interface DebugEventDomainService {
+    EventEntity createDebugApiEvent(
+        ExecutionContext executionContext,
```
Comment (ytvnr) [inline gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/core/debug/domain_service/DebugEventDomainService.java]: remove execution context

File: gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/core/debug/domain_service/DebugEventDomainService.java [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+        Set<String> environmentsIds,
```
Comment (ytvnr) [inline gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/core/debug/domain_service/DebugEventDomainService.java]: can I pass one and only envId ?

File: gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/infra/domain_service/debug/DebugEventDomainServiceLegacyWrapper.java [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+import io.gravitee.rest.api.model.EventType;
+import io.gravitee.rest.api.service.common.ExecutionContext;
+import java.util.Map;
+import java.util.Set;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+@RequiredArgsConstructor
+@Service
+public class DebugEventDomainServiceLegacyWrapper implements DebugEventDomainService {
```
Comment (ytvnr) [inline gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/infra/domain_service/debug/DebugEventDomainServiceLegacyWrapper.java]: get rid of it and directly call the eventCrudService

**Review by jgiovaresco — COMMENTED** [review]
Good job 💪  I will deploy a test env to run some test

**Review by jgiovaresco — APPROVED** [review]

File: gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/core/policy/domain_service/PolicyDomainService.java [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.gravitee.apim.core.policy.domain_service;
+
+public interface PolicyDomainService {
```
Comment (jgiovaresco) [inline gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/core/policy/domain_service/PolicyDomainService.java]:
```suggestion
public interface PolicyValidationDomainService {
```

File: gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/core/policy/domain_service/PolicyDomainService.java [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+    String validateAndSanitizePolicyConfiguration(String policyName, String configuration);
```
Comment (jgiovaresco) [inline gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/core/policy/domain_service/PolicyDomainService.java]:
```suggestion
    String validateAndSanitizeConfiguration(String policyName, String configuration);
```

**Review by jgiovaresco — APPROVED** [review]

**Pull request merged**